src/theme-context-provider.tsx

```diff
@@ -111,7 +111,7 @@ const ThemeContextProvider: React.FC<Props> = ({theme, children, as}) => {
                 title1: {...defaultTextPresetsConfig.title1, ...theme.skin.textPresets?.title1},
                 indicator: {...defaultTextPresetsConfig.indicator, ...theme.skin.textPresets?.indicator},
             },
-            Link: theme.Link ?? AnchorLink,
+            Link: getMisticaLinkComponent(theme.Link),
```

Comment:
This could cause potential remount issues because we are creating new Link components at render time.
It won't be an issue if the `theme` object reference is stable becase we have an useMemo here, but if someone does something like `<ThemeContextProvider theme={{/*define the theme inline here*/}}>` they'll have problems.
(I'm seeing we already have a warning for this in the README)

Reply:
This is somewhat out of scope of this task but perhaps a solution to this problem would be to store the theme inside a ref and don't react to changes. If someone wants a dynamic provider, they can force remounts with a key update. WDYT?

Reply:
Commented in teams, leaving this as is for now